--- condor.pm.bak	2008-04-30 16:43:40.660000000 -0500
+++ condor.pm	2008-05-01 09:10:36.421000000 -0500
@@ -347,6 +347,9 @@
     print SCRIPT_FILE "Input = " . $description->stdin() . "\n";
     print SCRIPT_FILE "Log = " . $self->{condor_logfile} . "\n";
     print SCRIPT_FILE "log_xml = True\n";
+# Patched by the VDT
+    print SCRIPT_FILE "+GratiaJobOrigin=\"GRAM\"\n";
+# End VDT patch
     print SCRIPT_FILE "#Extra attributes specified by client\n";
     print SCRIPT_FILE "$submit_attrs_string\n";
 
@@ -553,8 +556,11 @@
 
         if($num_abort > 0)
         {
-            # Don't delete the condor log when not reporting DONE
-            return Globus::GRAM::Error::SYSTEM_CANCELLED();
+            if ( ${self}->{individual_condor_log} ) {
+                $self->log_to_gratia();
+                unlink($self->{condor_logfile});
+            }
+            $state = Globus::GRAM::JobState::FAILED;
         }
         elsif($num_done == $description->count())
         {
@@ -620,17 +626,8 @@
 }
 
 # Patched by the VDT
-my $condor_version_number = 0;
 sub log_to_gratia
 {
-    unless ($condor_version_number) {
-        my $condor_version_string = `condor_version 2>/dev/null`;
-        $condor_version_number = 
-        join("",
-             map { m&^(\d+)&?sprintf("%03d", $1):"000" }
-             ($condor_version_string =~ m!^\$CondorVersion: ([^\.]+)\.([^\.]+)\.([^\.]+)!s));
-        return 1 unless ($condor_version_number < 6009000);
-    }
     my $self = shift;
     my $log_filename = $self->{condor_logfile};
 
@@ -663,19 +660,4 @@
     return 1; # Should return a proper Globus success code
 }
 
-sub cache_cleanup
-{
-    my $self = shift;
-    my $description = $self->{JobDescription};
-
-    if ( ${self}->{individual_condor_log} ) {
-        $self->log("Deleting Condor user log");
-        $self->log_to_gratia();
-        unlink($self->{condor_logfile});
-    }
-
-    return $self->SUPER::cache_cleanup($self);
-}
-
-
 1;
