--- globus-job-manager-script-real.pl.bak	2007-12-30 11:13:26.000000000 -0600
+++ globus-job-manager-script-real.pl	2008-01-28 16:24:46.688000000 -0600
@@ -36,6 +36,9 @@
 use Globus::GRAM::Error;
 use Globus::GRAM::JobDescription;
 
+eval "require Globus::GRAM::JobManagerGratia"; # Gratia patch to save DN/FQAN on submit
+my $gratia_callout = $@?0:1; # Don't make the callout if the file is not there.
+
 my($manager_name, $argument_file, $command) = (undef, undef, undef);
 
 GetOptions('manager-name|m=s' => \$manager_name,
@@ -105,6 +111,10 @@
 else
 {
     $manager->respond($result);
+    # Save cert information for Gratia accounting if appropriate.
+    if ($gratia_callout and ($command eq 'submit') and (exists $result->{JOB_ID})) {
+      Globus::GRAM::JobManagerGratia::gratia_save_cert_info($manager, $result->{JOB_ID});
+    }
 }
 
 sub fail
