diff -r -u urCollector-2006-06-13/urCollector.pl urCollector.mod/urCollector.pl
--- urCollector-2006-06-13/urCollector.pl	2006-06-13 03:02:45.000000000 -0500
+++ urCollector.mod/urCollector.pl	2006-08-24 02:53:49.819763200 -0500
@@ -1536,6 +1536,7 @@
 
 	# compute difference to UTC:
 	$tstamp = $tstamp + ($hourDiff * 3600) + ($minDiff * 60);
+	$tzStr = "Z";
     }
     else {
 	print "Warning: Wrong timezone format: $tzStr - cannot convert timestamp\n";
@@ -1635,8 +1636,9 @@
 	$remainingSecs = $remainingSecs - ($minutes * 60);
 	#print LOGFILE "days=$days; hours=$hours; minutes=$minutes; remainingSecs=$remainingSecs\n";
 	if ($days > 0) {
-	    $timePeriodString = $timePeriodString."$days"."DT";
+	    $timePeriodString = $timePeriodString."$days"."D";
 	}
+       $timePeriodString = $timePeriodString."T";
 	if ($hours > 0) {
 	    $timePeriodString = $timePeriodString."$hours"."H";
 	}
Only in urCollector.mod: urCollector.pl~
Only in urCollector.mod: urCreator.exe
diff -r -u urCollector-2006-06-13/urwg.cpp urCollector.mod/urwg.cpp
--- urCollector-2006-06-13/urwg.cpp	2006-06-13 03:00:42.000000000 -0500
+++ urCollector.mod/urwg.cpp	2006-08-23 15:50:43.860363200 -0500
@@ -58,7 +58,7 @@
 string urwg_UsageRecord::xml()	
 {
 	string outBuff;
-	outBuff = "<UsageRecord>\n";
+	outBuff = "<UsageRecord xmlns=\"http://www.gridforum.org/2003/ur-wg\" xmlns:urwg=\"http://www.gridforum.org/2003/ur-wg\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://www.gridforum.org/2003/ur-wg file:///u:/OSG/urwg-schema.11.xsd\">\n";
 	outBuff += composeRecordIdentity();
 	outBuff += composeJobIdentity();
 	outBuff += composeUserIdentity();
@@ -841,7 +841,10 @@
 	string buff;
 	buff = "<urwg:UserIdentity>\n";
 	buff += tagAdd("urwg:LocalUserId", userIdentity.LocalUserId);
-	buff += tagAdd("ds:KeyInfo", userIdentity.KeyInfo);
+       vector<attribute> attributes;
+       attribute attrBuff = {"xmlns:ds","http://www.w3.org/2000/09/xmldsig#"};        
+       attributes.push_back(attrBuff);
+	buff += tagAdd("ds:KeyInfo", userIdentity.KeyInfo,attributes);
 	buff +="</urwg:UserIdentity>\n";
 	return buff;
 }
diff -r -u urCollector-2006-06-13/xmlUtil.cpp urCollector.mod/xmlUtil.cpp
--- urCollector-2006-06-13/xmlUtil.cpp	2006-06-13 03:03:00.000000000 -0500
+++ urCollector.mod/xmlUtil.cpp	2006-08-24 02:45:52.293113600 -0500
@@ -206,13 +206,13 @@
         }
         if ( content == "" )
         {
-                //no Element value attached, end with \>
-                buff += "\\>\n";
+                //no Element value attached, end with />
+                buff += "/>\n";
         }
         else
         {
                 //Element Value is present 
-                buff +=">\n" + content + "\n</" +tag+ ">\n";
+                buff +=">" + content + "</" +tag+ ">\n";
         }
         return buff;
 }
@@ -269,7 +269,8 @@
         int seconds = result.rem;
         string buffer = "P";
         if ( days != 0 )
-                buffer += int2string(days) + "DT";
+                buffer += int2string(days) + "D";
+        buffer += "T";
         if ( hours != 0 )
                 buffer += int2string(hours) + "H";
         if ( minutes != 0 )
