--- urCollector.pl.104f	2009-08-26 12:05:56.000000000 -0500
+++ urCollector.pl	2009-09-03 10:41:15.000000000 -0500
@@ -984,6 +984,8 @@
 
     if ($userVo ne "") {
 	$cmd .= "\"UserVOName=$userVo\" ";
+    } elsif (exists $urAcctlogInfo{account} and $urAcctlogInfo{account} ne "") {
+	$cmd .= "\"UserVOName=$urAcctlogInfo{account}\" ";
     }
 
     if ($fqan ne "") {
@@ -1331,6 +1333,9 @@
 			$urAcctlogInfo{group} = $1;
       next;
 		}
+		if ( $record_field =~ /^account=(.*)$/o) {
+			$urAcctlogInfo{account} = $1;
+		}     
 		if ( $record_field =~ /^jobname=(.*)$/o) {	
 			$urAcctlogInfo{jobName} = $1;
       next;
