--- ../pbs-lsf-0.30b-2/urCollector.pl	2007-12-14 16:16:00.000000000 -0600
+++ ./urCollector.pl	2008-01-22 16:36:09.000000000 -0600
@@ -389,7 +389,7 @@
 sub processLrmsLogFile {
 
     my $filename = $_[0];
-    # my $newestF = $_[1];
+    my $newestF = $_[1];
 
     my $startJob = $_[2];
     my $startTimestamp = $_[3];
@@ -424,7 +424,11 @@
     my $firstJobId = 1;
 
     my $line = "";
+    my $line_counter = 0;
     while ($line = <LRMSLOGFILE>) {
+      ++$line_counter;
+      # Ignore the first record: pick it up next time
+      $newestF and $line_counter == 1 and next;
 
 	if ($useGGFformat) {
 	    %urGridInfo = ();    # reset grid-related info
