--- ./urCollector.pl~	2009-03-02 11:47:33.000000000 -0600
+++ ./urCollector.pl	2009-03-02 11:47:22.000000000 -0600
@@ -34,7 +34,7 @@
 POSIX::sigaction(&POSIX::SIGHUP, $actionHUP);
 POSIX::sigaction(&POSIX::SIGINT, $actionInt);
 POSIX::sigaction(&POSIX::SIGTERM, $actionInt);
-my $urCreatorExecutable = "$URCOLLECTOR_LOC/libexec/urCreator";
+my $urCreatorExecutable = "$URCOLLECTOR_LOC/urCreator";
 
 my $onlyOneIteration = 0; # default is run as daemon!
 my $useCElog = 1;   # default is use the CE's map: grid job <-> local job
