--- pbs-lsf-franklingrid/urCollector.pl	2008-12-15 16:18:22.000000000 -0600
+++ pbs-lsf-unl/urCollector.pl	2009-01-20 14:23:09.000000000 -0600
@@ -1288,11 +1288,11 @@
       $urAcctlogInfo{queue}=$1;
       next;
     }
-    if ( $record_field =~ /^resources_used.cput=.*?(\d\d):(\d\d):(\d\d)$/o) {
+		if ( $record_field =~ /^resources_used.cput=.*?(\d+):(\d+):(\d+)$/o) {
       $urAcctlogInfo{cput}= $3 + $2*60 + $1*3600;
       next;
     }
-    if ( $record_field =~ /^resources_used.walltime=*?(\d\d):(\d\d):(\d\d)$/o) {
+		if ( $record_field =~ /^resources_used.walltime=*?(\d+):(\d+):(\d+)$/o) {
       $urAcctlogInfo{walltime}= $3 + $2*60 + $1*3600;
       next;
     }
