--- urCollector.pl~	2009-01-20 15:28:47.000000000 -0600
+++ urCollector.pl	2009-01-20 15:29:33.000000000 -0600
@@ -1288,11 +1288,11 @@
       $urAcctlogInfo{queue}=$1;
       next;
     }
-		if ( $record_field =~ /^resources_used.cput=.*?(\d\d):(\d\d):(\d\d)$/o) {
+		if ( $record_field =~ /^resources_used.cput=.*?(\d+):(\d+):(\d+)$/o) {
 			$urAcctlogInfo{cput}= $3 + $2*60 + $1*3600;
       next;
 		}
-		if ( $record_field =~ /^resources_used.walltime=*?(\d\d):(\d\d):(\d\d)$/o) {
+		if ( $record_field =~ /^resources_used.walltime=*?(\d+):(\d+):(\d+)$/o) {
 			$urAcctlogInfo{walltime}= $3 + $2*60 + $1*3600;
       next;
 		}
